tests/qp_encoding_test.phpt

```diff
@@ -0,0 +1,98 @@
+--TEST--
+qp comprehensive test
+--SKIPIF--
+--FILE--
+<?php
+error_reporting(E_ALL); // ignore E_STRICT
+
+include("Mail/mimePart.php");
+
+/**
+ * Convenience function to make qp encoded output easier to verify
+ *
+ * @param string $text Input text to be encoded and printed
+ * @param int $begin Start character to visibly print from
+ * @param int $end Stop character to visibly print to
+ * @param bool $special_chars Convert character such as linebreaks
+ *     etc. to visible replacements.
+ * @param int $break Line length before soft break
+ *
+ */
+function debug_print($text, $begin=False, $end=False, $special_chars=True, $break=76) {
+    $begin = $begin ? $begin : strlen($text);
+    $end = $end ? $end : strlen($text);
+
+    for ($i=$begin; $i <= $end; $i++) {
+        $input = substr($text, 0, $i);
+        $output = Mail_mimePart::quotedPrintableEncode($input, $break);
+
+        printf("input: %02d: %s\n", strlen($input), $input);
+
+        $lines = explode("\r\n", $output);
+        for($j=0; $j < count($lines); $j++) {
+            $line = $lines[$j];
+            if ($j + 1 < count($lines) && $special_chars) {
+                $line_vis = str_replace("\t", '\t', $line).'\r\n';
```

Comment:
Would be nice to have this str_replace() used for printing the input too.

Reply:
It makes it easier to read when tabs are involved, but with newlines it's now one long string...

But ehh, it's six of one, half a dozen of the other... Added the change.